src/JsonRpcEngine.ts

```diff
@@ -54,9 +72,24 @@ export class JsonRpcEngine extends SafeEventEmitter {
 
   private _middleware: JsonRpcMiddleware<unknown, unknown>[];
 
-  constructor() {
+  private readonly _notificationHandler?: JsonRpcNotificationHandler<unknown>;
+
+  /**
+   * Constructs a {@link JsonRpcEngine} instance.
+   *
+   * @param options - Options bag.
+   * @param options.notificationHandler - A function for handling JSON-RPC
+   * notifications. A JSON-RPC notification is defined as a JSON-RPC request
+   * without an `id` property. If this option is _not_ provided, notifications
+   * will be treated the same as requests. If this option _is_ provided,
+   * notifications will be passed to the handler function without touching
+   * the engine's middleware stack. This function **should not** throw or
```

Comment:
The `**should not**` here appearly correctly but is obviously connected to the above so I guess they should be consistent, whatever you decide to do.